tools/iree-run-module-main.cc

```diff
@@ -221,6 +229,9 @@ extern "C" int main(int argc, char** argv) {
         "[ERROR] unexpected positional argument (expected none)."
         " Did you use pass a flag with a single dash ('-')?"
         " Use '--' instead.\n");
+#if IREE_USE_MPI
```

Comment:
I recommend moving the `exit_code` definition up, and rather than returning early and calling `MPI_Finalize` on each branch, set the exit code, and return only once, at the end, after calling `MPI_Finalize`.